sql/core/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowWriter.scala

```diff
@@ -55,6 +55,12 @@ object ArrowWriter {
       case (DoubleType, vector: NullableFloat8Vector) => new DoubleWriter(vector)
       case (StringType, vector: NullableVarCharVector) => new StringWriter(vector)
       case (BinaryType, vector: NullableVarBinaryVector) => new BinaryWriter(vector)
+      case (DateType, vector: NullableDateDayVector) => new DateWriter(vector)
+      case (TimestampType, vector: NullableTimeStampMicroTZVector)
+        // TODO: Should be able to access timezone from vector
+        if field.getType.isInstanceOf[ArrowType.Timestamp] &&
+          field.getType.asInstanceOf[ArrowType.Timestamp].getTimezone != null =>
```

Comment:
Do we need this `if`?

Reply:
It can probably be removed.  My original thinking was to ensure that the ArrowWriter was created with a valid timezone id and not a null or empty string.  I think it's better handled in `toArrowSchema` which takes an Option now.